app/src/org/commcare/dalvik/application/CommCareApplication.java

```diff
@@ -348,17 +355,103 @@ private void setRoots() {
     }
 
     private int initializeAppResources() {
-        //There should be exactly one of these for now
+        // Before we try to initialize a new app, check if any existing apps were left in a
+        // partially deleted state, and finish uninstalling them if so
         for (ApplicationRecord record : getGlobalStorage(ApplicationRecord.class)) {
-            if (record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
-                //We have an app record ready to go
+            if (record.getStatus() == ApplicationRecord.STATUS_DELETE_REQUESTED) {
+                record.uninstall(this);
+            }
+        }
+        
+        // There may now be multiple app records in storage, because of multiple apps support.
+        // Since this method is called from onCreate of CommCareApplication, current behavior is
+        // just to pick the first app in the list to initialize
+        for(ApplicationRecord record : getGlobalStorage(ApplicationRecord.class)) {
+            if(record.getStatus() == ApplicationRecord.STATUS_INSTALLED) {
                 return initializeAppResources(new CommCareApp(record));

```

Comment:
@ctsims - Actually the current app is accessed via CommCareApplication._().getCurrentApp(), which is itself set by a call to initializeAppResources(CommCareApp app). So there's no notion of a "current" app until a call to that method has occurred. But introducing the notion of a "favorite" app could be an interesting way of solving this- Have a mechanism in the app manager for setting one app to be the "favorite" at all times, which basically means "This is the app I'll use most often and it should be automatically booted up first." And if they don't choose to set that, then we can select the first?